interfaces/acados_template/acados_template/acados_casadi_ocp_solver.py

```diff
@@ -103,22 +107,22 @@ def create_casadi_nlp_formulation(cls, ocp: AcadosOcp) -> Tuple[dict, dict]:
         for i in range(solver_options.N_horizon):
             # add dynamics constraints
             if solver_options.integrator_type == "DISCRETE":
-                g.append(xtraj[:, i+1] - f_discr_fun(xtraj[:, i], utraj[:, i], ptraj[:, i], model.p_global))
+                g.apppend(xtraj_node[i+1] - f_discr_fun(xtraj_node[i], utraj_node[i], ptraj_node[i], model.p_global))
             elif solver_options.integrator_type == "ERK":
-                g.append(xtraj[:, i+1] - f_discr_fun(xtraj[:, i], utraj[:, i], solver_options.time_steps[i]))
+                g.append(xtraj_node[i+1] - f_discr_fun(xtraj_node[i], utraj_node[i], solver_options.time_steps[i]))
```

Comment:
not related to this PR, but the ERK integrator might also depend on `p` and `p_global` I guess?

Reply:
yes, tracked in https://github.com/acados/acados/issues/1525